### PR TITLE
Add multi cut file support

### PR DESCRIPTION
**Pull request by ddonukis**
## Description
Porting the multiple cut files support from the cog-shop repo.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/power-ops-sdk/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [pyproject.toml](https://github.com/cognitedata/power-ops-sdk/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by doctrino — APPROVED** [review]
Looks good, same change as I saw in the cogshop repo yesterday.

File: cognite/powerops/preprocessor/cogreader.py [hunk 1]
```diff
@@ -130,7 +104,7 @@ def get_extra_files_metadata(self) -> List[dict[str, str]]:
             return []
 
     @staticmethod
-    def file_metadata_to_dict(file_metadata):
+    def file_metadata_to_dict(file_metadata) -> dict[str, Union[str, int]]:
```
Comment (doctrino) [inline cognite/powerops/preprocessor/cogreader.py:107]: if you do `from __future__ import annotations` at the top you are porting in 3.10 type hints so you can write.
```suggestion
    def file_metadata_to_dict(file_metadata) -> dict[str, str | int]:
```

File: cognite/powerops/utils/retry/api.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    """
+    def decor(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            return caller(f, *args, **kwargs)
+        return wrapper
+    return decor
+
+
+def __retry_internal(f, exceptions=Exception, tries=-1, delay=0, max_delay=None, backoff=1, jitter=0,
```
Comment (doctrino) [inline cognite/powerops/utils/retry/api.py:26]:
```suggestion
def _retry_internal(f, exceptions=Exception, tries=-1, delay=0, max_delay=None, backoff=1, jitter=0,
```
Single underscore to indicate private, double underscore is to indicate variable mangling in Python

File: cognite/powerops/utils/retry/api.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                     logger=logging_logger):
+    """
+    Executes a function and retries it if it failed.
+
+    :param f: the function to execute.
+    :param exceptions: an exception or a tuple of exceptions to catch. default: Exception.
+    :param tries: the maximum number of attempts. default: -1 (infinite).
```
Comment (doctrino) [inline cognite/powerops/utils/retry/api.py:33]: Lets use numpy style for docs

File: cognite/powerops/utils/retry/api.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                _delay += jitter
+
+            if max_delay is not None:
+                _delay = min(_delay, max_delay)
+
+
+def retry(exceptions=Exception, tries=-1, delay=0, max_delay=None, backoff=1, jitter=0, logger=logging_logger):
+    """Returns a retry decorator.
+
+    :param exceptions: an exception or a tuple of exceptions to catch. default: Exception.
```
Comment (doctrino) [inline cognite/powerops/utils/retry/api.py:70]: numpy style

File: cognite/powerops/utils/retry/api.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    return retry_decorator
+
+
+def retry_call(f, fargs=None, fkwargs=None, exceptions=Exception, tries=-1, delay=0, max_delay=None, backoff=1,
+               jitter=0,
+               logger=logging_logger):
+    """
+    Calls a function and re-executes it if it failed.
+
+    :param f: the function to execute.
```
Comment (doctrino) [inline cognite/powerops/utils/retry/api.py:98]: and here.

**Review by doctrino — COMMENTED** [review]
A few minor things

**Review by Juliamg — APPROVED** [review]
lgtm

**Pull request merged**